program-runtime/src/invoke_context.rs

```diff
@@ -369,8 +369,13 @@ impl<'a> InvokeContext<'a> {
             }
 
             self.pre_accounts = Vec::with_capacity(instruction_accounts.len());
-            let mut work = |_index_in_instruction: usize,
-                            instruction_account: &InstructionAccount| {
+
+            for (index_in_instruction, instruction_account) in
+                instruction_accounts.iter().enumerate()
+            {
+                if index_in_instruction != instruction_account.index_in_callee {
```

Comment:
This logic is confusing, having a hard time wrapping my head around it

Reply:
If the `index_in_instruction` (position of this account in the parameters) is not `index_in_callee` (first occurrence of this accounts key) then it must be duplicate and is skipped.

Reply:
I see, makes more sense now, a good comment here would help